src/dmd/parse.d

```diff
@@ -8919,15 +8919,32 @@ final class Parser(AST) : Lexer
         endtok = token.value == TOK.leftBracket ? TOK.rightBracket : TOK.rightParentheses;
 
         nextToken();
+
+        int errorsBefore = errorCount;
+        bool newErrors = false;
         while (token.value != endtok && token.value != TOK.endOfFile)
         {
             auto arg = parseAssignExp();
             arguments.push(arg);
             if (token.value == endtok)
                 break;
-            check(TOK.comma);
+            if(token.value != TOK.comma)
+            {
+                error("found `%s` when expecting `%s` or `%s`", token.toChars(), Token.toChars(TOK.comma), Token.toChars(endtok));
```

Comment:
`Token.toChars(TOK.comma)` can be replaced by a literal, directly in the specifier, cant it ?